### PR TITLE
Add  Open to gateway API

### PR DESCRIPTION
**Pull request by LovisaLugnegard**
To be able to create an open command in reva.

### CODE REVIEW

**Comment by glpatcern** [comment]
As discussed at the telco, it looks OK to me but I'd like @SamuAlfageme to give his green light as you know better this part.

File: cs3/gateway/v1beta1/gateway_api.proto [hunk 1]
```diff
@@ -308,6 +309,8 @@ service GatewayAPI {
 
   // Creates a new ocm share.
   rpc CreateOCMCoreShare(cs3.ocm.core.v1beta1.CreateOCMCoreShareRequest) returns (cs3.ocm.core.v1beta1.CreateOCMCoreShareResponse);
+  // Returns the iframe url.
+  rpc Open(cs3.app.provider.v1beta1.OpenRequest) returns (cs3.app.provider.v1beta1.OpenResponse);
```
Comment (SamuAlfageme) [inline cs3/gateway/v1beta1/gateway_api.proto:313]: I would go for a more descriptive name of what the method actually `Open`s  (e.g. `OpenFileInAppProvider()`) and a bit of context on where the iframe URL comes from.

Comment (SamuAlfageme) [reply cs3/gateway/v1beta1/gateway_api.proto:313]: Also, is it a part of the "`OCM CORE`" as the block comment mentions on top? Does it make sense to create a new block for "`APP PROVIDERS`"?

Comment (LovisaLugnegard) [reply cs3/gateway/v1beta1/gateway_api.proto:313]: Good points, will update this!

Comment (LovisaLugnegard) [reply cs3/gateway/v1beta1/gateway_api.proto]: Thank you for the comments. I agree that `OpenFileInAppProvider` is a good name! WHat do you think @glpatcern?

Comment (glpatcern) [reply cs3/gateway/v1beta1/gateway_api.proto]: Yes! for an API, I like it being verbose.
Also the return URL - being it iframed or not - should reflect the fact that it comes from an external Application. `AppProviderURL`? Any better suggestion?

**Comment by labkode** [comment]
@LovisaLugnegard any update on this one?

File: cs3/gateway/v1beta1/gateway_api.proto (hunk 1 above)
Comment (LovisaLugnegard) [reply cs3/gateway/v1beta1/gateway_api.proto]: 👍 I'll update this

**Comment by LovisaLugnegard** [comment]
@labkode Yes, it's updated now

**Comment by glpatcern** [comment]
(I don't seem to have the ability to "approve" but looks good to me: as discussed, having a more self-explanatory API is excellent here)

**Comment by labkode** [comment]
@ishank011are these changes breaking the recently added provider refactoring?

**Comment by ishank011** [comment]
@labkode no they won't as long as we don't update the go-cs3apis version in reva without adding the corresponding changes. Also, since this is a breaking change, I'd prefer that there's not much of a delay between merging this and the changes in reva, since otherwise anyone wanting to make changes to cs3apis would be blocked.

Also, you might want to update the OpenRequest definition again since I reverted that.

**Comment by labkode** [comment]
@LovisaLugnegard in the Open request the field `  storage.provider.v1beta1.ResourceInfo resource_info = 2;
` should be a reference.

**Comment by LovisaLugnegard** [comment]
@labkode Thanks for finding this, I had forgot to push :)

**Pull request merged**